server/src/internalClusterTest/java/org/elasticsearch/index/engine/ThreadPoolMergeSchedulerStressTestIT.java

```diff
@@ -82,7 +82,9 @@ public static class TestEnginePlugin extends Plugin implements EnginePlugin {
         void allowAllMerging() {
             // even when indexing is done, queued and backlogged merges can themselves trigger further merging
             // don't let this test be bothered by that, and simply let all merging run unhindered
-            runMergeSemaphore.release(Integer.MAX_VALUE - initialRunMergesCount);
+            if (runMergeSemaphore.availablePermits() < 10_000) {
```

Comment:
How can it have more than 10K available already? Should we instead assert it has not?

Reply:
It can't have more than 10K, or Integer.MAX_VALUE / 2, 99999, etc...
It's just strange to fudge any sort of non-obvious value, but on the other hand, the shenanigans required to avoid that are not worth it. I'll fudge a value, and drop a comment.